core/src/main/scala/codesearch/core/search/Search.scala

```diff
@@ -204,11 +204,11 @@ object Search {
       result: CodeSnippet
   )
 }
-sealed trait ErrorMessage
-final case class SearchByIndexResult(lists: List[String], error: ErrorMessageFromSearchByIndex)
-final case class ErrorMessageFromSearchByIndex(error: String = "") extends ErrorMessage
+sealed trait Response
+final case class SearchByIndexResult(lists: List[String], error: ErrorResponse)
+final case class ErrorResponse(message: String) extends Response
 final case class CSearchPage(
     data: Seq[PackageResult],
     total: Int,
-    errorMessage: ErrorMessageFromSearchByIndex
-)
+    error: ErrorResponse
```

Comment:
It's very confusing.
Remove the `error` field from `CSearchPage` case class and use `ErrorResponse` independently only for errors.
Also remove the `SearchByIndexResult` case class because now we check only that regexp is correct without stdout error handling.

Reply:
But how i will return ErrorResponse in case of error to html? Now we return CSearchPage and how i can return another case class instead of CSearchPage?

Reply:
@Kabowyad 
`ErrorResponse` and `CSearchPage` case classes is your ADT over `Response` trait. Based on this you can use `Response` trait and patmatch it in Twirl.